bundles/org.openhab.binding.millheat/src/main/java/org/openhab/binding/millheat/internal/MillheatHandlerFactory.java

```diff
@@ -0,0 +1,113 @@
+/**
+ * Copyright (c) 2010-2019 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.millheat.internal;
+
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Hashtable;
+import java.util.Map;
+import java.util.Set;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.eclipse.jetty.client.HttpClient;
+import org.eclipse.smarthome.config.discovery.DiscoveryService;
+import org.eclipse.smarthome.core.thing.Bridge;
+import org.eclipse.smarthome.core.thing.Thing;
+import org.eclipse.smarthome.core.thing.ThingTypeUID;
+import org.eclipse.smarthome.core.thing.ThingUID;
+import org.eclipse.smarthome.core.thing.binding.BaseThingHandlerFactory;
+import org.eclipse.smarthome.core.thing.binding.ThingHandler;
+import org.eclipse.smarthome.core.thing.binding.ThingHandlerFactory;
+import org.eclipse.smarthome.io.net.http.HttpClientFactory;
+import org.openhab.binding.millheat.internal.discovery.MillheatDiscoveryService;
+import org.openhab.binding.millheat.internal.handler.MillheatAccountHandler;
+import org.openhab.binding.millheat.internal.handler.MillheatHeaterHandler;
+import org.openhab.binding.millheat.internal.handler.MillheatHomeHandler;
+import org.openhab.binding.millheat.internal.handler.MillheatRoomHandler;
+import org.osgi.framework.ServiceRegistration;
+import org.osgi.service.component.annotations.Activate;
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Reference;
+
+/**
+ * The {@link MillheatHandlerFactory} is responsible for creating things and thing
+ * handlers.
+ *
+ * @author Arne Seime - Initial contribution
+ */
+@NonNullByDefault
+@Component(configurationPid = "binding.millheat", service = ThingHandlerFactory.class)
+public class MillheatHandlerFactory extends BaseThingHandlerFactory {
+    private HttpClient httpClient;
+    private Map<ThingUID, ServiceRegistration<?>> discoveryServiceRegs = new HashMap<>();
```

Comment:
in the mean time, since my last comment, the preferred way discovery services are being registered has been changed to implement the discovery as ` ThingServiceHandler`.

Reply:
Should be ok for now and can be refactored after the merge.